### PR TITLE
Removing shortcut methods [WIP]

### PR DESCRIPTION
**Pull request by krainboltgreene**
We really don't need these, since the regular way of creating these objects is pretty simple already.

### CODE REVIEW

**Comment by alexdowad** [comment]
Small is beautiful! :)

File: README.md [hunk 1]
```diff
@@ -197,7 +197,7 @@ list.tail                    # => Hamster.list(2, 3)
 To add to a list, you use `List#cons`:
 
 ``` ruby
-original = Hamster.list(1, 2, 3)
+original = Hamster::List(1, 2, 3)
```
Comment (dubek) [inline README.md]: This should be brackets: `Hamster::List[1, 2, 3]`

This shows the awkward situation in which `Vector` `Hash` `Set` are classes (and therefore instances are constructed with `#new`), but `List` is a module... Maybe topic for another discussion.

**Comment by alexdowad** [comment]
Is there any reason why we can't define a class method called "new" on List?

**Comment by dubek** [comment]
@alexdowad I believe adding `List.new` will make things even more difficult to understand:

``` ruby
$ irb -Ilib -rhamster
2.2.1 :001 > my_list = Hamster::List[1,2,3]    # Or, per suggestion:   Hamster::List.new(1,2,3)
 => Hamster::List[1, 2, 3] 
2.2.1 :002 > my_list.class
 => Hamster::Cons 
```

**Comment by alexdowad** [comment]
OK, fair enough. Agreed.

**Comment by krainboltgreene** [comment]
Yeah, the List class has all sorts of weird shortcut methods that involve
Cons, and I'm fine leaving it that way.

**Comment by dubek** [comment]
OK sure. The README should be fixed though (line 200) to square brackets.

**Comment by krainboltgreene** [comment]
This is a WIP.

**Comment by alexdowad** [comment]
I just removed `Hamster.vector`, including updating all the documentation and specs, in f444320. Will push a commit which removes `Hamster.hash` next.

**Comment by krainboltgreene** [comment]
<3

**Comment by alexdowad** [comment]
`Hamster.hash` removed in c102c1f. Now the `Hamster` module can be used as a key in a `Hash` or similar collection. (I doubt anybody ever tried this before.)

Question: should `Hamster::Hash.[]` be able to take a default block? With `Hamster.hash` removed, people will have to use `Hamster::Hash.new` if they want to create a `Hash` with a default block.

**Comment by alexdowad** [comment]
`Hamster.set` removed in 8be58e3.

**Comment by alexdowad** [comment]
`Hamster.sorted_set` removed in 64261e6.

**Comment by alexdowad** [comment]
`Hamster.deque` removed in 809b6af.

**Comment by alexdowad** [comment]
`Hamster.list` removed in aba0a4b.

**Comment by alexdowad** [comment]
`.mutable_hash`, `.mutable_set`, and `.mutable_queue` shortcut methods have also been removed. This is now done.

**Pull request closed without merging**